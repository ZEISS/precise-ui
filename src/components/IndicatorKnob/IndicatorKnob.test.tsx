import * as React from 'react';
import * as enzyme from 'enzyme';
import { IndicatorKnob } from './';

describe('<IndicatorKnob />', () => {
  it('should render an empty <IndicatorKnob>', () => {
    const wrapper = enzyme.shallow(<IndicatorKnob />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an active <IndicatorKnob>', () => {
    const wrapper = enzyme.shallow(<IndicatorKnob active />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <IndicatorKnob> in red', () => {
    const wrapper = enzyme.shallow(<IndicatorKnob color="red" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a disabled <IndicatorKnob>', () => {
    const wrapper = enzyme.shallow(<IndicatorKnob disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an active disabled <IndicatorKnob>', () => {
    const wrapper = enzyme.shallow(<IndicatorKnob active disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the <IndicatorKnob> with explicit x and y coordinates', () => {
    const wrapper = enzyme.shallow(<IndicatorKnob x={1} y={1} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <IndicatorKnob> at the top', () => {
    const wrapper = enzyme.shallow(<IndicatorKnob y={0} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <IndicatorKnob> on the left', () => {
    const wrapper = enzyme.shallow(<IndicatorKnob x={0} />);
    expect(wrapper).toMatchSnapshot();
  });
});
