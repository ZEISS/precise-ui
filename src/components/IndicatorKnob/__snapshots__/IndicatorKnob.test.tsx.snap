// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndicatorKnob /> should render <IndicatorKnob> at the top 1`] = `
<styled.div
  style={
    Object {
      "backgroundColor": undefined,
      "left": "50%",
      "top": "0%",
    }
  }
/>
`;

exports[`<IndicatorKnob /> should render <IndicatorKnob> on the left 1`] = `
<styled.div
  style={
    Object {
      "backgroundColor": undefined,
      "left": "0%",
      "top": "50%",
    }
  }
/>
`;

exports[`<IndicatorKnob /> should render a disabled <IndicatorKnob> 1`] = `
<styled.div
  disabled={true}
  style={
    Object {
      "backgroundColor": undefined,
      "left": "50%",
      "top": "50%",
    }
  }
/>
`;

exports[`<IndicatorKnob /> should render an active <IndicatorKnob> 1`] = `
<styled.div
  active={true}
  style={
    Object {
      "backgroundColor": undefined,
      "left": "50%",
      "top": "50%",
    }
  }
/>
`;

exports[`<IndicatorKnob /> should render an active disabled <IndicatorKnob> 1`] = `
<styled.div
  active={true}
  disabled={true}
  style={
    Object {
      "backgroundColor": undefined,
      "left": "50%",
      "top": "50%",
    }
  }
/>
`;

exports[`<IndicatorKnob /> should render an empty <IndicatorKnob> 1`] = `
<styled.div
  style={
    Object {
      "backgroundColor": undefined,
      "left": "50%",
      "top": "50%",
    }
  }
/>
`;

exports[`<IndicatorKnob /> should render the <IndicatorKnob> in red 1`] = `
<styled.div
  style={
    Object {
      "backgroundColor": "red",
      "left": "50%",
      "top": "50%",
    }
  }
/>
`;

exports[`<IndicatorKnob /> should render the <IndicatorKnob> with explicit x and y coordinates 1`] = `
<styled.div
  style={
    Object {
      "backgroundColor": undefined,
      "left": "100%",
      "top": "100%",
    }
  }
/>
`;
